api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -80,6 +81,33 @@ public interface DeleteOrphanFiles extends Action<DeleteOrphanFiles, DeleteOrpha
    */
   DeleteOrphanFiles executeDeleteWith(ExecutorService executorService);
 
+  /**
+   * Pass a mode for handling the files that cannot be determined if they are orphan
+   * @param mode mode for handling files that cannot be determined if they are orphan
+   * @return this for method chaining
+   */
+  default DeleteOrphanFiles prefixMismatchMode(PrefixMisMatchMode mode) {
+    throw new UnsupportedOperationException(this.getClass().getName() + " does not implement prefixMismatchMode");
+  }
+
+  /**
+   * Pass a list of schemes to be considered equivalent when finding orphan files
+   * @param equivalentSchemes list of equivalent schemes
+   * @return this for method chaining
+   */
+  default DeleteOrphanFiles equivalentSchemes(List<String> equivalentSchemes) {
```

Comment:
I think we should use `Map<String, String>` for scheme and authority equivalency.

Reply:
@aokolnychyi 
Do you mean create a make a map like s3a: s3, s3n: s3, s3a: s3 (similarly for authorities) ie map each scheme in the equiavalent list to same name in this case `s3`
And during normalization, look up the map and populate the scheme and authorities, so that comparisons are just `equals` instead of having to search in a list?

Reply:
Yeah, I mean what @karuppayya described above. I feel that's more flexible and easier to consume in the action.